src/definition.jl

```diff
@@ -51,6 +59,50 @@ function module_definition_location(mod::Module)
             var"end" = Position(; line = line - 1, character = Int(typemax(Int32)))))
 end
 
+get_location(bind::Tuple{JL.SyntaxTree, URI}) =  Location(; uri = bind[2], range = get_source_range(bind[1]))
+
+function definition_target_methods(objtyp)
```

Comment:
As discussed earlier, this PR will be refactored so that the caller side handles case-by-case branching for each type of `objtyp`, but I'll comment on this for future reference.

In implementations like this one that dispatch based on lattice objects returned by the compiler, the compiler generates dynamic dispatch because the type of `objtyp` cannot be statically determined from the caller side either.
In such cases, using `@nospecialize` can enable static dispatch.